core/src/main/java/com/graphhopper/routing/VirtualEdgeIterator.java

```diff
@@ -213,7 +213,11 @@ public EdgeIteratorState setName(String name) {
 
     @Override
     public String toString() {
-        return edges.toString();
+        if (current >= 0 && current < edges.size()) {
+            return "virtual edge: " + edges.get(current) + ", all: " + edges.toString();
+        } else {
+            return "virtual edge: (invalid)" + ", all: " + edges.toString();
```

Comment:
Shouldn't this better throw an IllegalStateException?

Reply:
No I don't think so because `current=-1` is the state the iterator is in before calling `next()` for the first time.